pkg/apis/apiserver/organization_REST.go

```diff
@@ -106,19 +125,253 @@ func (o *OrganizationREST) NewList() runtime.Object {
+		return nil, false, err
+	}
+
+	u, err := ConvertToV1Alpha1Unstructured(newObj.(*Organization), o.scheme)
+	if err != nil {
+		return nil, false, err
+	}
+
+	var subresource []string
+	if a.GetSubresource() != "" {
+		subresource = append(subresource, a.GetSubresource())
+	}
+	u, err = o.dynamicRI.Update(ctx, u, *options, subresource...)
+	if err != nil {
+		return nil, false, err
+	}
+
+	retObj, err := ConvertFromV1Alpha1Unstructured(u, o.scheme)
+	if err != nil {
+		return nil, false, err
+	}
+	return retObj, false, nil
+}
+
+func (o *OrganizationREST) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	visible, err := o.isVisible(ctx, name, "delete")
+	if err != nil {
+		return nil, false, err
+	}
+	if !visible {
+		return nil, false, apierrors.NewNotFound(qualifiedResource, name)
+	}
+	obj, err := o.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, false, err
+	}
+	if err := deleteValidation(ctx, obj); err != nil {
+		return obj, false, err
+	}
+	err = o.dynamicRI.Delete(ctx, name, *options)
+	return obj, false, err
+}
+
+func (o *OrganizationREST) DeleteCollection(ctx context.Context, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions, listOptions *internalversion.ListOptions) (runtime.Object, error) {
+	opts := &metav1.ListOptions{}
+	if err := o.scheme.Convert(listOptions, opts, nil); err != nil {
+		return nil, err
+	}
+	err := o.dynamicRI.DeleteCollection(ctx, *options, *opts)
+	return nil, err
+}
+
+func (o *OrganizationREST) Watch(ctx context.Context, options *internalversion.ListOptions) (watch.Interface, error) {
+	opts := &metav1.ListOptions{}
+	if err := o.scheme.Convert(options, opts, nil); err != nil {
+		return nil, err
+	}
+	wi, err := o.dynamicRI.Watch(ctx, *opts)
+	if err != nil {
+		return nil, err
+	}
+	res := make(chan watch.Event)
+	pw := watch.NewProxyWatcher(res)
+	go func() {
+		defer wi.Stop()
+		defer close(res)
+		for {
+			select {
+			case <-pw.StopChan():
+				return
+			case ev := <-wi.ResultChan():
+				if ev.Type == watch.Error {
+					res <- ev
+					return
+				}
+				org, err := ConvertFromV1Alpha1Unstructured(ev.Object.(*unstructured.Unstructured), o.scheme)
+				if err != nil {
+					res <- internalErrorWatchEvent(err)
+					return
+				}
+				ev.Object = org
+				visible, err := o.isVisible(ctx, org.Name, "watch")
+				if err != nil {
+					res <- internalErrorWatchEvent(err)
+					return
+				}
+				if visible {
+					res <- ev
+				}
+			}
+		}
+	}()
+	return pw, nil
+}
+
+func internalErrorWatchEvent(err error) watch.Event {
+	return watch.Event{
+		Type: watch.Error,
+		Object: &metav1.Status{
+			Status:  "error",
+			Message: err.Error(),
+			Reason:  metav1.StatusReasonInternalError,
+			Code:    http.StatusInternalServerError,
+		},
+	}
+}
+
+func (o *OrganizationREST) isVisible(ctx context.Context, organizationName string, verb string) (bool, error) {
```

Comment:
This function is ok for this PR, but we should refactor the filtering to using the `members` in the Organization.Status.Members. and output the Organization that the user is in.

Reply:
Sure, will do that in #27   as well as add proper e2e tests there.